### ISSUE TITLE
release-notes - not clear what to do for going from 4.0.0 to 4.0.3

### ISSUE THREAD

**Issue by Martin-Weiss** [SUSE/doc-caasp#644]
Just updated a cluster from 4.0.0 to 4.0.3 and realized that the dex hint in 4.0.2 about "If you have not applied 4.0.1" is still valid and required but not listed in 4.0.3 required actions.
But the step in 4.0.1 about "The gangway image that shipped with SUSE CaaS Platform 4.0 must be updated manually by performing the following steps:" is not required / valid anymore.

So in case a customer does not apply each step when it gets released and skips i.e. from 4.0.0 to 4.0.3 - the procedure of the upgrade following the release notes is not really easy to understand or possible to follow.

I am not sure how to fix this problem in the release notes - probably the latest version - i.e. 4.0.3 needs to cover all steps required from 4.0.0 to 4.0.3 with skipping intermediate and not required steps?

**Comment by r0ckarong** [comment]
@kkaempf Our maintenance releases are incremental right?
We can mention this specifically in the next set of release notes and ask people to go through the instructions. We included all steps since 4.0 in the new RN format.

Writing on consistent instruction is beyong the scope of release notes and is "migration" which to me belongs in the admin guide.

**Comment by Martin-Weiss** [comment]
zypper up from 4.0.0 to 4.0.3 is not incremental.

**Comment by kkaempf** [comment]
> @kkaempf Our maintenance releases are incremental right?

afaik, yes.

> Writing one consistent instruction is beyond the scope of release notes and is "migration" which to me belongs in the admin guide.

Ack.

**Comment by kkaempf** [comment]
> zypper up from 4.0.0 to 4.0.3 is not incremental.

Don't do `zypper up` on a node. Period. Let `skuba` handle this.

**Comment by Martin-Weiss** [comment]
> 
> 
> > zypper up from 4.0.0 to 4.0.3 is not incremental.
> 
> Don't do `zypper up` on a node. Period. Let `skuba` handle this.

Probably this was not clear enough - zypper up needs to be done on the admin machine to get the latest skuba and then we run the skuba commands to upgrade the cluster and the cluster addons. 

Based on staging - the upgrade will be from 4.0.0 to i.e. 4.0.3 skipping 4.0.1 and 4.0.2.

In this case it is not clear what steps to follow when reading through the release notes or the admin guide. This is why I created this issue.

**Comment by Martin-Weiss** [comment]
> > Don't do `zypper up` on a node. Period. Let `skuba` handle this.

In case this is such important - how do we prevent an admin from doing this and how can support detect that a customer did this "wrong"?

Btw. what might happen in case the admin machine already has a newer patchlevel (as zypper up ; reboot was done manually), then the customer runs "skuba node upgrade apply / skuba addon upgrade apply".. (change staging at 8:00am, update admin machine 9:00am and run node/addon upgrade 10:00am where the skuba-update.timer starts around midnight somewhat later)?
In skuba cluster status I can not even see if the nodes already have 4.0.3... - but I believe there was a enhancement on the way?

**Comment by kkaempf** [comment]
> > > Don't do `zypper up` on a node. Period. Let `skuba` handle this.
> 
> In case this is such important - how do we prevent an admin from doing this

You can't prevent stupid. We would need to remove the `root` account.

>  and how can support detect that a customer did this "wrong"?

That's an interesting question. `zypper` has an option to pass a `transaction id`:
```
> zypper -h
  ...
    --userdata <STRING>     User defined transaction id used in history and plugins.
  ...
```
which will appear in the `zypper.log`. `skuba` could set this id to indicated "zypper was called from skuba" in the zypper.log.

**Comment by kkaempf** [comment]
Created a [skuba issue](https://github.com/SUSE/skuba/issues/904) to track this.

**Comment by Martin-Weiss** [comment]
Probably I missed it but the information about "do not use zypper up/dup/patch.. on CaaSP nodes" seems to be missing in the admin guide or at least I can not find it..

**Comment by r0ckarong** [comment]
> Probably I missed it but the information about "do not use zypper up/dup/patch.. on CaaSP nodes" seems to be missing in the admin guide or at least I can not find it..

Until this issue I was not aware of this limitation. I will add something to the admin guide.

**Comment by r0ckarong** [comment]
Closed with #645
